src/bridge/SequencerInbox.sol

```diff
@@ -101,28 +104,62 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
         rollup = newRollup;
     }
 
-    function getTimeBounds() internal view virtual returns (TimeBounds memory) {
-        TimeBounds memory bounds;
-        if (block.timestamp > maxTimeVariation.delaySeconds) {
-            bounds.minTimestamp = uint64(block.timestamp - maxTimeVariation.delaySeconds);
+    /// @inheritdoc ISequencerInbox
+    function totalDelayedMessagesRead() public view returns (uint256) {
+        return bridge.totalDelayedMessagesRead();
+    }
+
+    function getTimeBounds() internal view virtual returns (IBridge.TimeBounds memory) {
+        IBridge.TimeBounds memory bounds;
+        (
+            uint64 delayBlocks_,
+            uint64 futureBlocks_,
+            uint64 delaySeconds_,
+            uint64 futureSeconds_
+        ) = maxTimeVariationInternal();
+        if (block.timestamp > delaySeconds_) {
+            bounds.minTimestamp = uint64(block.timestamp) - delaySeconds_;
         }
-        bounds.maxTimestamp = uint64(block.timestamp + maxTimeVariation.futureSeconds);
-        if (block.number > maxTimeVariation.delayBlocks) {
-            bounds.minBlockNumber = uint64(block.number - maxTimeVariation.delayBlocks);
+        bounds.maxTimestamp = uint64(block.timestamp) + futureSeconds_;
+        if (block.number > delayBlocks_) {
+            bounds.minBlockNumber = uint64(block.number) - delayBlocks_;
         }
-        bounds.maxBlockNumber = uint64(block.number + maxTimeVariation.futureBlocks);
+        bounds.maxBlockNumber = uint64(block.number) + futureBlocks_;
         return bounds;
     }
 
-    /// @inheritdoc ISequencerInbox
-    function removeDelayAfterFork() external {
-        if (!_chainIdChanged()) revert NotForked();
-        maxTimeVariation = ISequencerInbox.MaxTimeVariation({
-            delayBlocks: 1,
-            futureBlocks: 1,
-            delaySeconds: 1,
-            futureSeconds: 1
-        });
+    function maxTimeVariation() public view returns (ISequencerInbox.MaxTimeVariation memory) {
+        (
+            uint64 delayBlocks_,
+            uint64 futureBlocks_,
+            uint64 delaySeconds_,
+            uint64 futureSeconds_
+        ) = maxTimeVariationInternal();
+
+        return
+            ISequencerInbox.MaxTimeVariation({
+                delayBlocks: delayBlocks_,
+                futureBlocks: futureBlocks_,
+                delaySeconds: delaySeconds_,
+                futureSeconds: futureSeconds_
+            });
+    }
```

Comment:
To preserve backwards compatibility between versions can you make this return the old 4xuint256 struct? I think this is only called with eth_call so there is no need to save gas.

Reply:
Hmm I suppose we can fix it like this https://github.com/OffchainLabs/nitro-contracts/commit/f3cfc60cd9cedb7b715dd3d4aad69bec43125e09 (not committed to this branch yet)

Reply:
I think this will work since it's going back to the old signature https://github.com/OffchainLabs/nitro-contracts/pull/68/files/773e1ec74b404098f8eead510c727bfc471afad7#diff-f49bf9a48371b4b787c61abac8f18edfa65baa865c69e140b5a67d430070f245L81-L89

Reply:
Merged [f3cf](https://github.com/OffchainLabs/nitro-contracts/commit/f3cfc60cd9cedb7b715dd3d4aad69bec43125e09)